apps/pipeline-visualization/src/mermaid_builder.ts

```diff
@@ -0,0 +1,138 @@
+// SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+import { strict as assert } from 'assert';
+
+import {
+  BlockDefinition,
+  PipelineDefinition,
+} from '@jvalue/jayvee-language-server';
+
+export interface MermaidOptions {
+  mermaidFile: string;
+  styleFile: string;
+  compositeBlocks: boolean;
+  properties: boolean;
+}
+
+export class MermaidBuilder {
+  protected linkLines: string[] = [];
+  protected nodeLines: string[] = [];
+  protected styleLines: string[] = [];
+  protected indent = 0;
+
+  constructor(protected options: MermaidOptions) {}
+
+  escapeHtml(unsafe: string) {
```

Comment:
(Improvement, not required) Make methods that a caller doesn't need or shouldn't use `protected`.